### PR TITLE
Improve gradle build configuration

### PR DESCRIPTION
**Pull request by maximilian-galanis**
- [x] Rework gradle structure to make language frontends fully optional
- [x] Remove the now redundant 'experimental' Tags and Annotations from the experimental frontends
- [x] Adapt cpg-neo4j and cpg-console to deal with optional frontends 
- [x] Adapt CI/CD to set the now necessary properties to enable the experimental frontends
- [x] Switch from dependabot to renovate (renovate supports gradle version catalogs)
- [x] Test CI/CD
- [x] Check Dokka
- [x] Test new feature variants in cpg-console & cpg-neo4j

### CODE REVIEW

**Comment by oxisto** [comment]
Feature request: Can we add a small script like `configure_frontends.sh` that does a quick Y/N check which frontends to you want to enable and sets the corresponding flags in the properties file? That would be an extremely nice workflow for developers.

**Comment by maximilian-galanis** [comment]
> Feature request: Can we add a small script like `configure_frontends.sh` that does a quick Y/N check which frontends to you want to enable and sets the corresponding flags in the properties file? That would be an extremely nice workflow for developers.

Yes, that should be pretty easy to include. I just think that maybe this problem is better solved with good documentation in the README file, which informs users about optional frontends and how to toggle them on/off.

With the suggested configure_frontends.sh we would still have to tell the user about the optional frontends but instead of telling her directly how to toggle them, we would tell her to run a shell script, which toggles them for her.

**Comment by oxisto** [comment]
> > Feature request: Can we add a small script like `configure_frontends.sh` that does a quick Y/N check which frontends to you want to enable and sets the corresponding flags in the properties file? That would be an extremely nice workflow for developers.
> 
> Yes, that should be pretty easy to include. I just think that maybe this problem is better solved with good documentation in the README file, which informs users about optional frontends and how to toggle them on/off.
> 
> With the suggested configure_frontends.sh we would still have to tell the user about the optional frontends but instead of telling her directly how to toggle them, we would tell her to run a shell script, which toggles them for her.

Yes good documentation is one thing but this would be a nice convenience feature for us internally as well I think.

File: cpg-core/src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt [hunk 1]
```diff
@@ -385,6 +389,35 @@ private constructor(
             return this
         }
 
+        /**
+         * Loads and registers an additional [Language] based on a fully qualified class name (FQN).
+         */
+        @Throws(ConfigurationException::class)
+        fun registerLanguage(className: String): Builder {
+            val loadedClass: Class<*>
```
Comment (oxisto) [inline cpg-core/src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt:397]: For consistency can we use `KClass`?

**Comment by oxisto** [comment]
While we are at it. Could you also check the GitHub actions warnings?
<html>
<body>


code-complianceNode.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, Fraunhofer-AISEC/codyze-action, actions/checkout
--
code-complianceThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
build-cpgo-osxNode.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, actions/setup-go, actions/setup-java, actions/upload-artifact, actions/upload-artifact, actions/setup-java, actions/checkoutShow more
build-cpgo-osxThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
build-cpgo-osxThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
build-cpgo-osxThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildNode.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, actions/setup-java, actions/setup-python, actions/setup-node, actions/setup-go, actions/cache, actions/upload-artifact, actions/cache, actions/setup-node, actions/setup-python, actions/setup-java, actions/checkoutShow more
buildThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
buildThe `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
build: .github#L1macOS-latest pipelines will use macOS-12 soon. For more details, see https://github.com/actions/runner-images/issues/6384


</body>
</html>[code-compliance](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736083140)
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, Fraunhofer-AISEC/codyze-action, actions/checkout
[code-compliance](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736083140#step:3:11)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build-cpgo-osx](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736083319)
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, actions/setup-go, actions/setup-java, actions/upload-artifact, actions/upload-artifact, actions/setup-java, actions/checkout
[build-cpgo-osx](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736083319#step:4:24)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build-cpgo-osx](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736083319#step:4:25)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build-cpgo-osx](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736083319#step:4:26)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298)
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, actions/setup-java, actions/setup-python, actions/setup-node, actions/setup-go, actions/cache, actions/upload-artifact, actions/cache, actions/setup-node, actions/setup-python, actions/setup-java, actions/checkout
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:9:22)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:4:8)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:3:23)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:3:24)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:3:25)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:3:34)
The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:3:48)
The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:8:11)
The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:8:17)
The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build](https://github.com/Fraunhofer-AISEC/cpg/actions/runs/3439142799/jobs/5736118298#step:8:19)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[build: .github#L1](https://github.com/Fraunhofer-AISEC/cpg/pull/968/files#annotation_5955930554)
macOS-latest pipelines will use macOS-12 soon. For more details, see https://github.com/actions/runner-images/issues/6384

File: cpg-core/src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt (hunk 1 above)
Comment (maximilian-galanis) [reply cpg-core/src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt]: Done

**Pull request merged**